source/PlayerInfo.cpp

```diff
@@ -866,14 +866,19 @@ Ship *PlayerInfo::Flagship()
 // Determine which ship is the flagship and return the shared pointer to it.
 const shared_ptr<Ship> &PlayerInfo::FlagshipPtr()
 {
+	bool foundFlagship = false;
 	if(!flagship)
 	{
 		for(const shared_ptr<Ship> &it : ships)
-			if(!it->IsParked() && it->GetSystem() == system && it->CanBeFlagship())
+			if(!it->IsParked() && it->GetSystem() == system && it->CanBeFlagship() && !foundFlagship)
 			{
 				flagship = it;
-				break;
```

Comment:
Why did you remove the break instead of just moving it 2 lines lower, afterthe new operations you've added in this if?

Reply:
Seems good to me then